filters/filters.go

```diff
@@ -151,6 +151,15 @@ type Filter interface {
 	Response(FilterContext)
 }
 
+// FilterCloser are Filters that need to cleanup resources after
+// filter teminiation. For example Filters, that create a goroutine
+// for some reason need to cleanup their goroutine or they would leak
+// goroutines.
+type FilterCloser interface {
+	Filter
+	Close()
```

Comment:
Maybe it should be `Close() error` (i.e. io.Closer) before its too late - we can always ignore error but it would be hard to introduce it later.

Reply:
With returned error you can decide to ignore it or handle. Without error there is no way to handle an error if any.

Reply:
fixed

Reply:
fixed